# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: 0552c0b4a079
# Full Hash: 0552c0b4a079a695edfca100df68c4ad82481e58
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Refactor away RtcpEventObserver in favor of MediaEvents. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123223
# ==============================================================================

diff -r 877a0fcc6da9 -r 0552c0b4a079 dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Mon Aug 16 15:57:10 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Thu Aug 19 11:49:29 2021 +0200
@@ -16,7 +16,6 @@
 #include "mozilla/media/MediaUtils.h"
 #include "VideoTypes.h"
 #include "WebrtcVideoCodecFactory.h"
-#include "RtcpEventObserver.h"
 #include "nsTArray.h"
 #include "mozilla/dom/RTCRtpSourcesBinding.h"
 
@@ -199,6 +198,9 @@
 
   virtual bool HasCodecPluginID(uint64_t aPluginID) const = 0;
 
+  virtual MediaEventSource<void>& RtcpByeEvent() = 0;
+  virtual MediaEventSource<void>& RtcpTimeoutEvent() = 0;
+
   virtual void DeliverPacket(rtc::CopyOnWriteBuffer packet,
                              PacketType type) = 0;
 
@@ -207,8 +209,6 @@
   virtual Maybe<RefPtr<AudioSessionConduit>> AsAudioSessionConduit() = 0;
   virtual Maybe<RefPtr<VideoSessionConduit>> AsVideoSessionConduit() = 0;
 
-  virtual void SetRtcpEventObserver(RtcpEventObserver* observer) = 0;
-
   virtual webrtc::Call::Stats GetCallStats() const = 0;
 
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(MediaSessionConduit)